internal/util/fetch/fetch.go

```diff
@@ -223,3 +233,20 @@ func ClonerUsingGitExec(ctx context.Context, repoSpec *git.RepoSpec) error {
 	}
 	return nil
 }
+
+// checkPackageTags looks up most specific tag for the package at the provided
+// pkgPath.
+func checkPackageTags(pkgPath, ref string, upstreamRepo *gitutil.GitUpstreamRepo) string {
+	// Check if we have a ref in the upstream that matches the package-specific
+	// reference. If we do, we use that reference.
+	ps := strings.Split(pkgPath, "/")
```

Comment:
https://golang.org/pkg/path/filepath/#SplitList

Reply:
The package path provided in the Kptfile is OS agnostic, so we should not use the OS specific `filepath` package here.